tower-service/src/lib.rs

```diff
@@ -139,48 +137,84 @@ use std::task::{Context, Poll};
 ///     }
 /// }
 ///
+/// // The error returned if processing a request timed out
+/// #[derive(Debug)]
+/// pub struct Expired;
+///
+/// impl fmt::Display for Expired {
+///     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+///         write!(f, "expired")
+///     }
+/// }
+///
+/// impl Error for Expired {}
+///
+/// // We can implement `Service` for `Timeout<T>` if `T` is a `Service`
 /// impl<T, Request> Service<Request> for Timeout<T>
 /// where
 ///     T: Service<Request>,
 ///     T::Future: 'static,
-///     T::Error: From<Expired> + 'static,
-///     T::Response: 'static
+///     T::Error: Into<Box<dyn Error>> + 'static,
```

Comment:
```suggestion
///     T::Error: Into<Box<dyn Error + Send + Sync>> + 'static,
```

Because trait obj are a PITA